OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDisabled.cs

```diff
@@ -0,0 +1,60 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Applies a condition to the actor when it is disabled.",
+		"This is a temporary shim to help migration away from the legacy IDisable code")]
```

Comment:
Having an actor-level disabled state interacts badly with the rest of our code (see e.g. your #7807).  We started the process or removing it quite a while ago, but progress stalled because we couldn't combine upgrade types / conditions in the ways we needed to.